src/Symfony/Component/AstGenerator/AstGeneratorChain.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AstGenerator;
+
+/**
+ * Generator delegating the generation to a chain of generators.
+ *
+ * @author Joel Wurtz <joel.wurtz@gmail.com>
+ */
+class AstGeneratorChain implements AstGeneratorInterface
```

Comment:
> This is weird Symfony internal rule: to keep members private by default, but in the same time do not require final: there is no point to open them for inheritance then.

What exactly is wrong with hooking in on `parent::someMethod()`? Ie why should it enforce composition over inheritance? Isnt this matter of choice/design...